tests/utils.go

```diff
@@ -457,3 +455,52 @@ func deletePVC(pvcname string) {
 	status := IsPVCDeletedEventually(pvcname)
 	gomega.Expect(status).To(gomega.Equal(true), "while trying to get deleted pvc")
 }
+
+func verifyPVForPVC(shouldExist bool, pvcName string) {
+	pvList, err := PVClient.List(metav1.ListOptions{})
+	gomega.Expect(err).To(
+		gomega.BeNil(),
+		"while listing PV for PVC: %{s}",
+		pvcName,
+	)
+
+	shouldPVExist := gomega.BeFalse()
+	if shouldExist {
+		shouldPVExist = gomega.BeTrue()
+	}
+
+	ginkgo.By("verifying PV for PVC exists or not")
+	var matchingPVName string
+	matchingPVExists := false
+	for _, pv := range pvList.Items {
```

Comment:
this looks ok. Can't we query the pv before deleting the pvc and then delete the pvc and verify the pv?

Reply:
yes, we can . It will just verify that the PV is present for the PVC.  We are already checking if the PVC is in Bound state, so it will be just an additional check to the same.